src/components/TaskItem/TaskItem.tsx

```diff
@@ -0,0 +1,75 @@
+import styles from "./TaskItem.module.scss";
+import Checkbox from "../Checkbox/Checkbox";
+import { useEffect, useRef, useState } from "react";
+import { MdDelete } from "react-icons/md";
+import { AiFillEdit } from "react-icons/ai";
+import IconButton from "../IconButton/IconButton";
+import clsx from "clsx";
+
+type TaskItemProps = {
+  message: string;
+  completed: boolean;
+  messageEditable: boolean;
+};
+
+const TaskItem = ({
+  message = "first task",
+  completed = false,
+  messageEditable = false,
+}: TaskItemProps) => {
+  const [title, setTitle] = useState(message);
+  const [checked, setChecked] = useState(completed);
+  const [isEditable, setIsEditbale] = useState(messageEditable);
+  const ref = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (ref?.current && isEditable) {
+      ref.current.focus();
+    }
+  }, [isEditable]);
+
+  return (
+    <div className={styles.TaskItem}>
+      <Checkbox checked={checked} onChange={() => setChecked(!checked)} />
+      <input
+        ref={ref}
+        className={clsx(styles.taskInput, {
+          [styles.taskInputEditable]: isEditable && !checked,
+          [styles.taskInputCompleted]: checked,
+        })}
+        type="text"
+        value={title}
+        onChange={(e) => setTitle(e.target.value)}
+        readOnly={checked}
+        onClick={() => {
+          setIsEditbale(true);
+        }}
+        onBlur={(e) => {
+          if (e.relatedTarget?.id === "editButton") {
+            return;
```

Comment:
what does e.relatedTarget do gere ? Why not e.target?

Reply:
this is the logic - to prevent setingState to false when client click on editButton after editing the title

Reply:
This is something we can discuss in details. Why you do not want to have that option to disable editing on either click outside or edit btn click? Did you consider to highlit edit button (either by outline or icon color change) to indicate Edit state?

Reply:
Yes, and this is what onBlur is doing, but editButton also changes state isEditable to !isEditable. I can not see better way to do it, than target if after onBlur was clicked editButton and stop changing state to false.
Highliting the editButton can be in my opinion  done by adding new class based on isEditable and checked state.

If You have some better ideas please let me know :)